### PR TITLE
Implement Users page backend

### LINKED ISSUE #68: Implement Users page
We have a GitHub [user entity](https://github.com/OWASP/Nest/blob/ad5a48f650b119cb30e40ec99ff926fa39d29af4/backend/apps/github/models/user.py#L9) model that's currently used in Projects page only. 

The idea is to create a standalone Users page that will be responsible for showing/searching users within GitHub OWASP organization. Each user entry will be clickable and should lead to another view showing a detailed user page (this is going to be covered in another issue).

The task involves both backend and frontend changes. Currently we don't have user index in Algolia. So as the first step for backend we need to index users (we have plenty of examples how to do that, e.g. [issues](https://github.com/OWASP/Nest/blob/ad5a48f650b119cb30e40ec99ff926fa39d29af4/backend/apps/github/index/issue.py#L10), [projects](https://github.com/OWASP/Nest/blob/ad5a48f650b119cb30e40ec99ff926fa39d29af4/backend/apps/owasp/index/project.py#L10), [chapters](https://github.com/OWASP/Nest/blob/ad5a48f650b119cb30e40ec99ff926fa39d29af4/backend/apps/owasp/index/chapter.py#L10)).

After creating the users index the data needs to be exposed via users search endpoint (it also needs to be implemented using existing examples, e.g. [project](https://github.com/OWASP/Nest/blob/ad5a48f650b119cb30e40ec99ff926fa39d29af4/backend/apps/owasp/api/search/project.py#L13)).

The frontend part of the task is to create a component for user entry and use it for populating Users page. After the [pagination component](https://github.com/OWASP/Nest/issues/56) is ready it'll be added to the users page as well.

### PR DESCRIPTION
**Pull request by Yashgupta9330**
Resolves #68
Made the backend of users page

### CODE REVIEW

File: backend/Makefile [hunk 1]
```diff
@@ -41,7 +41,7 @@ load-data:
 	@echo "Loading Nest data"
 	@cp backend/data/nest.json.gz backend/data/copy.json.gz
 	@gunzip backend/data/nest.json.gz
-	@CMD="poetry run python manage.py load_data" $(MAKE) exec-backend-command
+	@CMD="poetry run python manage.py load_data" && $(MAKE) exec-backend-command
```
Comment (arkid15r) [inline backend/Makefile]: Why this change?

File: backend/apps/github/api/user.py
Comment (arkid15r) [inline backend/apps/github/api/user.py:1]: We don't provide index data via API yet -- we use local DB data only. Please revert it.

File: backend/apps/github/index/user.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+from algoliasearch_django import AlgoliaIndex
+from algoliasearch_django.decorators import register
+
+from apps.common.index import IndexSynonymsMixin
+from apps.github.models import User
+
+
+@register(User)
+class UserIndex(AlgoliaIndex, IndexSynonymsMixin):
+    """User index for OWASP GitHub organization members."""
```
Comment (arkid15r) [inline backend/apps/github/index/user.py]:
```suggestion
    """User index."""
```

File: backend/apps/github/migrations/0015_remove_user_is_hireable_remove_user_twitter_username_and_more.py
Comment (arkid15r) [inline backend/apps/github/migrations/0015_remove_user_is_hireable_remove_user_twitter_username_and_more.py:1]: Could you take care of the migrations -- why you remove and then add fields back?

File: backend/apps/github/models/user.py [hunk 3]
```diff
@@ -1,46 +1,108 @@
...
+    GITHUB_NODE_ID_NOT_FOUND,
+    GITHUB_NODE_ID_REQUIRED,
+    GITHUB_USER_DATA_REQUIRED,
+)
 from apps.github.models.common import GenericUserModel, NodeModel
 
 
 class User(NodeModel, GenericUserModel, TimestampedModel):
-    """User model."""
+    """Model representing a GitHub user or organization."""
```
Comment (arkid15r) [inline backend/apps/github/models/user.py]: We have a separate [model](https://github.com/OWASP/Nest/blob/main/backend/apps/github/models/organization.py) for GitHub organizations

File: backend/apps/github/models/user.py [hunk 4]
```diff
@@ -1,46 +1,108 @@
...
+            models.Index(fields=["type"]),
+        ]
 
-    bio = models.TextField(verbose_name="Bio", max_length=1000, default="")
-    is_hireable = models.BooleanField(verbose_name="Is hireable", default=False)
-    twitter_username = models.CharField(verbose_name="Twitter username", max_length=50, default="")
+    bio = models.TextField(
+        verbose_name="Bio",
+        max_length=1000,
+        blank=True,
```
Comment (arkid15r) [inline backend/apps/github/models/user.py]: Why `blank` here?

File: backend/apps/github/models/user.py [hunk 5]
```diff
@@ -1,46 +1,108 @@
...
+        """Return the string representation of the user."""
         return f"{self.name or self.login}"
 
+    def clean(self):
+        """Validate required fields before saving the user."""
+        super().clean()
+        if not self.login:
+            raise ValidationError(GITHUB_LOGIN_REQUIRED)
+        if not self.node_id:
+            raise ValidationError(GITHUB_NODE_ID_REQUIRED)
```
Comment (arkid15r) [inline backend/apps/github/models/user.py]: Did you see cases when it was absent?

File: backend/apps/github/models/user.py [hunk 6]
```diff
@@ -1,46 +1,108 @@
...
+            "location": "location",
+            "login": "login",
+            "name": "name",
+            "created_at": "created_at",
+            "updated_at": "updated_at",
             "bio": "bio",
-            "is_hireable": "hireable",
-            "twitter_username": "twitter_username",
+            "type": "type",
+            "public_repositories_count": "public_repos",
```
Comment (arkid15r) [inline backend/apps/github/models/user.py]: This is done in [GenericUserModel](https://github.com/OWASP/Nest/blob/4082718b8bf470557c38313595295724465254bc/backend/apps/github/models/common.py#L45)

File: backend/data/copy.json.gz
Comment (arkid15r) [inline backend/data/copy.json.gz:1]: Please revert this.

File: backend/apps/github/models/user.py [hunk 7]
```diff
@@ -1,46 +1,108 @@
...
+        verbose_name="User type",
+        max_length=50,
+        default="User",
+        choices=[
+            ("User", "User"),
+            ("Organization", "Organization"),
+        ],
+        help_text="Distinguishes between User and Organization accounts",
+    )
+    public_repositories_count = models.PositiveIntegerField(
```
Comment (arkid15r) [inline backend/apps/github/models/user.py]: It already exists -- https://github.com/OWASP/Nest/blob/4082718b8bf470557c38313595295724465254bc/backend/apps/github/models/common.py#L26

File: backend/apps/github/models/user.py [hunk 8]
```diff
@@ -1,46 +1,108 @@
...
-    is_hireable = models.BooleanField(verbose_name="Is hireable", default=False)
-    twitter_username = models.CharField(verbose_name="Twitter username", max_length=50, default="")
+    bio = models.TextField(
+        verbose_name="Bio",
+        max_length=1000,
+        blank=True,
+        default="",
+        help_text="User's GitHub biography",
+    )
+    type = models.CharField(
```
Comment (arkid15r) [inline backend/apps/github/models/user.py]: As I mentioned -- users and orgs are separate entities.

**Review by arkid15r — CHANGES_REQUESTED** [review]
It seems a lot of my comments are based on your attempt to use User models for organization data. We have a separate model for that. 

Please address when you get a chance:

File: backend/apps/github/models/user.py (hunk 4 above)
Comment (Yashgupta9330) [reply backend/apps/github/models/user.py]: it is because bio is optional

**Review by arkid15r — APPROVED** [review]

**Pull request merged**